src/components/Faucet/Stripe.tsx

```diff
@@ -0,0 +1,227 @@
+import { useAccount } from 'wagmi'
+
+const STRIPE_PUBLIC_KEY = import.meta.env.STRIPE_PUBLIC_KEY
+const stripePromise = loadStripe(STRIPE_PUBLIC_KEY)
+
+type CheckoutFormProps = {
+  amount?: string
+  returnURL?: string
+}
+
+type StatusData = {
+  status: string | null
+  customer_email: string | null
+  faucet_package: string | undefined
+  recipient: string | null
+  quantity: number | null
+}
+
+export const CheckoutForm = ({ amount, returnURL }: CheckoutFormProps) => {
+  const { address } = useAccount()
+  const { client } = useClient()
+  const origin = window.location.origin
+
+  const fetchClientSecret = useCallback(async () => {
+    if (address) {
+      let uri = `${client.faucetService.url}/createCheckoutSession/${address}`
+      if (amount && !isNaN(parseInt(amount))) {
+        uri = `${uri}/${amount}`
+      }
+      // Create a Checkout Session
+      return await fetch(uri, {
+        method: 'POST',
+        body: JSON.stringify({
+          returnURL: returnURL || `${origin}/stripe/return`,
+          referral: origin,
+        }),
+      })
+        .then((res) => res.json())
+        .then((data) => data.clientSecret)
+    }
+    return await Promise.resolve('')
+  }, [address])
+
+  const options = { fetchClientSecret, clientSecret: '' }
+
+  return (
+    <Box id='checkout' mt={10} mb={24}>
+      <EmbeddedCheckoutProvider stripe={stripePromise} options={options}>
+        <EmbeddedCheckout />
+      </EmbeddedCheckoutProvider>
+    </Box>
+  )
+}
+
+type CheckoutReturnProps = {
+  sessionId: string
+}
+
+export const CheckoutReturn = ({ sessionId }: CheckoutReturnProps) => {
+  const { t } = useTranslation()
+  const { client, loaded: accountLoaded, account, fetchAccount } = useClient()
+  const toast = useToast()
+  const [status, setStatus] = useState<string | null>(null)
+  const [customerEmail, setCustomerEmail] = useState<string | null>('')
+  const [faucetPackage, setFaucetPackage] = useState<string | undefined>('')
+  const [recipient, setRecipient] = useState<string | null>('')
+  const [packageConsumed, setPackageConsumed] = useState(false)
+  const [aborted, setAborted] = useState(false)
+
+  // fetch the session status
+  useEffect(() => {
+    const abortController = new AbortController()
+    const signal = abortController.signal
+    const getStatus = async (): Promise<StatusData | null> => {
+      const res = await fetch(`${client.faucetService.url}/sessionStatus/${sessionId}`, { signal })
+      const data = await res.json()
+
+      if (data.faucet_package !== null) {
+        return data
+      }
+      if (!faucetPackage) {
+        return await getStatus()
+      }
+      return null
+    }
+    getStatus().then((data) => {
+      if (!data) return
+      setStatus(data.status)
+      setCustomerEmail(data.customer_email)
+      setFaucetPackage(data.faucet_package)
+      setRecipient(data.recipient)
+    })
+
+    return () => {
+      abortController.abort()
+    }
+  }, [])
```

Comment:
Kinda cumbersome to define a method that returns stuff and then use that return to set state values. Why not simply set state values during he function execution?

Also, why the infinite loop checking for a state value that won't be on the first run for sure??